### PR TITLE
Feature/golds 40 configure for routes service

### PR DESCRIPTION
**Pull request by rohitdobariyaleanspace**

### CODE REVIEW

File: testing/routes/processors/main.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    query = ""
+    page  = 0
+    size  = 10
+    sort  = ["name,asc"]
+  }
+}
+
+resource "leanspace_processors" "test_create_processor" {
+  file_path                            = var.path
+  version                              = "v1.0"
```
Comment (jerome-leanspace) [inline testing/routes/processors/main.tf:26]: Nit: no need for `v` it seems to be already displayed on the UI

File: examples/resources/leanspace_processors.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+variable "processor_path" {
+  type        = string
+  description = "The path to the processor's source code file (a .jar)."
+}
+
+resource "leanspace_processors" "test_create_processor" {
+  file_path                            = var.processor_path
+  version                              = "v1.0"
```
Comment (jerome-leanspace) [inline examples/resources/leanspace_processors.tf:8]: Nit: no need for v it seems to be already displayed on the UI

File: testing/routes/routes/main.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    size            = 10
+    sort            = ["name,asc"]
+  }
+}
+
+resource "leanspace_routes" "test_create_route" {
+  name                                 = "Terraform Route"
+  description                          = "This is a Route created through terraform!"
+  definition {
+    configuration   = "- route"
```
[the remark this answers is not in the document]
Comment (rohitdobariyaleanspace) [reply testing/routes/routes/main.tf]: Done

File: examples/resources/leanspace_routes.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+resource "leanspace_routes" "test_create_route" {
+  name                                 = "Terraform Route"
+  description                          = "This is a Route created through terraform!"
+  definition {
+    configuration   = "- route"
```
[the remark this answers is not in the document]
Comment (rohitdobariyaleanspace) [reply examples/resources/leanspace_routes.tf]: Done

File: testing/routes/processors/main.tf (hunk 1 above)
Comment (rohitdobariyaleanspace) [reply testing/routes/processors/main.tf]: Done

File: examples/resources/leanspace_processors.tf (hunk 2 above)
Comment (rohitdobariyaleanspace) [reply examples/resources/leanspace_processors.tf]: Done

File: README.md [hunk 5]
```diff
@@ -153,5 +153,9 @@ The available resources per service are:
   - members: it has three `leanspace_members` resources, created recursively.
   - service accounts: it has three `leanspace_service_accounts` resources, created recursively.  
   - teams: it has one `leanspace_teams` resource, created with the given members and policies.
+- routes:
+  - routes: it has one `leanspace_routes` resource, with basic filler data.
```
[the remark this answers is not in the document]
Comment (rohitdobariyaleanspace) [reply README.md]: Done

**Comment by jerome-leanspace** [comment]
2 things left:
- modifying the processor needs to be checked (it gives me a 404 because it tries to send the full payload with wrong mime content-type instead of json)
- adding an attached processor to the route in the example and testing

File: helper/helper.go [hunk 6]
```diff
@@ -67,6 +70,43 @@ func Ptr[T any](value T) *T {
 	return &value
 }
 
+func CustomEncoding(filePath string, data []byte) (io.Reader, string, error) {
```
Comment (N1ark) [inline helper/helper.go:73]: Maybe give this a more explicit name? Like `FileAndDataToMultipart` or something that explicitly says what the function does

**Comment by jerome-leanspace** [comment]
@rohitdobariyaleanspace @N1ark I've updated the PR with one commit that should fix what's remaining
Can you check?

**Review by jerome-leanspace — APPROVED** [review]
Please remember to update the minor version when releasing :)

**Comment by rohitdobariyaleanspace** [comment]
> @rohitdobariyaleanspace @N1ark I've updated the PR with one commit that should fix what's remaining Can you check?

All are good and ready to merge . I would like test locally cross check and merge this PR

**Pull request merged**